app/config/services.yml

```diff
@@ -1,83 +1,115 @@
+            tags: ['security.voter']
 
-    # Event Listeners are classes that listen to one or more specific events.
-    # Those events are defined in the tags added to the service definition.
-    # See http://symfony.com/doc/current/event_dispatcher.html#creating-an-event-listener
-    app.redirect_to_preferred_locale_listener:
-        class: AppBundle\EventListener\RedirectToPreferredLocaleListener
-        arguments: ['@router', '%app_locales%', '%locale%']
-        tags:
-            - { name: kernel.event_listener, event: kernel.request, method: onKernelRequest }
+        Twig_ExtensionInterface:
+            tags: ['twig.extension']
 
-    app.comment_notification:
-        class: AppBundle\EventListener\CommentNotificationListener
-        arguments: ['@mailer', '@router', '@translator', '%app.notifications.email_sender%']
-        # The "method" attribute of this tag is optional and defaults to "on + camelCasedEventName"
-        # If the event is "comment.created" the method executed by default is "onCommentCreated()".
-        tags:
-            - { name: kernel.event_listener, event: comment.created, method: onCommentCreated }
+        # The following settings will be appended to any local service definitions
+        # whose class matches this type. In this example, it will only apply to
+        # the "AppBundle\Controller\Admin\BlogController" service discovered by
+        # the "AppBundle\Controller\" rule below.
+        AppBundle\Controller\Admin\BlogController:
+            getters:
+                getSlugger: '@AppBundle\Utils\Slugger'
 
-    # Event subscribers are similar to event listeners but they don't need service tags.
-    # Instead, the PHP class of the event subscriber includes a method that returns
-    # the list of events listened by that class.
-    # See http://symfony.com/doc/current/event_dispatcher.html#creating-an-event-subscriber
-    app.requirements_subscriber:
-        class: AppBundle\EventListener\CheckRequirementsSubscriber
-        arguments: ['@doctrine.orm.entity_manager']
-        tags:
-            - { name: kernel.event_subscriber }
+    # This section enables to automatically register all classes found in the matching
+    # file paths and directories as services in the container. File and directory
+    # matching uses any valid glob pattern to create a white list of paths.
+    #
+    # In this example, the classes found in the following directories will be
+    # automatically registered as services:
+    #
+    # * src/AppBundle/EventListener/
+    # * src/AppBundle/Form/Type/
+    # * src/AppBundle/Security/
+    # * src/AppBundle/Twig/
+    # * src/AppBundle/Utils/
+    #
+    # The "resource" attribute can accept more specific glob patterns like
+    # "{EventListener/*Subscriber.php,Form/Type/*Type.php}".
+    AppBundle\:
+        # Register all classes in the src/AppBundle directory as services
+        resource: '../../src/AppBundle/{EventListener,Form/Type,Security,Twig,Utils}'
```

Comment:
Yes the current implementation is broken. I'm working on a PR to make form type services privatable.